packages/reports/addon/models/bard-request/request.js

```diff
@@ -459,6 +459,27 @@ export default Fragment.extend(Validations, {
     return this.removeSort(sortObj);
   },
 
+  /**
+   * Removes the sort from the sort array using the metric model and parameters
+   *
+   * @method removeSortMetricByModelAndParam
+   * @param {DS.Model} metricModel
+   * @param {Object} [parameters] - parameters [optional]
+   * @returns {Void}
+   */
+  removeSortMetricByModelAndParam(metricModel, parameters) {
+    if (parameters) {
+      let canonicalizedMetric = canonicalizeMetric({
+        metric: get(metricModel, 'name'),
+        parameters
+      });
+      this.removeSortByMetricName(canonicalizedMetric);
+    } else {
```

Comment:
because `removeSortByMetricName` finds by `canonicalName`, while we want to filter by the name and remove sorted metrics with all parameters. In other words the if block is called with specific parameter, the else block is called when you unselect the metric itself when it can have more than one param selected

Reply:
hmm.. makes sense. My concern is that the method name now can be a little confusing. its called `removeSortMetricWithParam` where parameters is an optional argument.

Reply:
https://github.com/yahoo/navi/pull/174/files#diff-c41a86855938e5b6dd5b8089ddb920caL458
One suggestion, can we update this method to remove all sorts with the same metricName.
Switch the `findBy` to `filterBy` and loop through the sorts and remove them.

Reply:
did you mean `removeSortByMetricName`?

Reply:
Yes, can we move the else logic there?

Reply:
Sounds like a good idea